js/internal/modules/modules.go

```diff
@@ -0,0 +1,52 @@
+/*
+ *
+ * k6 - a next-generation load testing tool
+ * Copyright (C) 2020 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package modules
+
+import (
+	"fmt"
+	"sync"
+)
+
+//nolint:gochecknoglobals
+var (
+	modules = make(map[string]interface{})
+	mx      sync.RWMutex
```

Comment:
nit: I am not certain having a Lock is truly necessary as the Register happens in init functions and I would argue should not happen at any other time. And the inits are by definition not concurrent so :shrug: 

Don't get me wrong ... it is probably better to have lock in the case that we ever decide to have Register outside the `init` functons. And I doubt the `RLock` significantly slows anything

Reply:
Good point, I wasn't aware `init()`s weren't concurrent.

I don't feel strongly about it to remove it, but if it's YAGNI then we might as well do so.

@na-- What do you think?

Reply:
I don't think it matters much - it shouldn't be needed, but it won't slow down anything either, so :man_shrugging: